### PR TITLE
fix ugrid reader

### PR DESCRIPTION
**Pull request by rajeeja**

### CODE REVIEW

**Comment by rajeeja** [comment]
Fixes #27 #28

**Review by michaelavs — APPROVED** [review]
I think this looks good, I was able to run the tests successfully locally! I do want to note that grid.py was having issues at first because I have edits to my version locally (I think from scrip PR/branch) and I think your version and mine were overlapping somewhat, so definitely wait for at least one more review before merging this just in case

File: uxarray/_ugrid.py
Comment (erogluorhan) [inline uxarray/_ugrid.py:56]: `var_names_dict` gets updated after leaving this ffunction, but for the sake of code readability, should we also return that in here:

`return xr_ds, var_names_dict`

and change Line 155 in grid.py as follows:

`self.ds, self.var_names_dict = _read_ugrid(self.filepath, self.var_names_dict)`?

I am not  sure which way to go though.

**Review by erogluorhan — COMMENTED** [review]
Great work overall, thanks! Please add the newly added functions into the docs as well as address inline comments.

File: uxarray/_ugrid.py
Comment (rajeeja) [reply uxarray/_ugrid.py]: fixed

File: uxarray/_ugrid.py
Comment (rajeeja) [reply uxarray/_ugrid.py]: makes sense. fixed.

File: uxarray/grid.py
Comment (rajeeja) [reply uxarray/grid.py]: done.

File: docs/internal_api/index.rst
Comment (erogluorhan) [inline docs/internal_api/index.rst:19]: Typo: "__init_ds_var_names__"

**Review by erogluorhan — COMMENTED** [review]
Just a single inline comment before approval

File: uxarray/helpers.py
Comment (erogluorhan) [inline uxarray/helpers.py:100]: Is this intended to be an internal or user API function? Also, please add it to corresponding docs index.

Comment (rajeeja) [reply uxarray/helpers.py:100]: This isn't used. Removing.

File: docs/internal_api/index.rst
Comment (rajeeja) [reply docs/internal_api/index.rst:19]: Fixed

**Review by erogluorhan — APPROVED** [review]

File: uxarray/helpers.py
Comment (dcherian) [inline uxarray/helpers.py:50]: xarray does cache datasets but it might be cleaner to just use `open_dataset` once like `ext_ds` above.

Comment (rajeeja) [reply uxarray/helpers.py:50]: List is for "xr.open_dataset(
                            filepath, mask_and_scale=False).filter_by_attrs(
                                node_coordinates=standard_name).keys()" There could be multiple keys, so taking the first ("[0]") from first element from the list.

Comment (dcherian) [reply uxarray/helpers.py:50]: Thanks. I meant tthat you could
```
ds = xr.open_dataset(filepath, mask_and_scale=False)
```
and then
```
ds.filter_by_attrs(node_coordinates=standard_name).keys()
```
since it gets opened repeatedly (it doesn't actually, because xarray caches the open file)

Comment (rajeeja) [reply uxarray/helpers.py:50]: gotcha, thanks.

**Pull request merged**
